DAT-427 - Change sort order selector to links

madelinekosse opened the pull request:
When tab-selecting the "order by" selector on the search page, the page is automatically reloaded when you select an option. We want users to be able to see the options without applying them so I have changed the selector to a list of links, with the currently active sort order highlighted in pink.

madelinekosse committed:
DAT-427 - Change sort order selector to links

When tab-selecting the "order by" selector on the search page, the page is
automatically reloaded when you select an option. We want users to be able to
see the options without applying them so I have changed the selector to a list
of links, with the currently active sort order highlighted in pink.

kjoshi reviewed (changes requested):
Is there a default sort order when `sort` isn't set in the query params?

I think it's relevance?
![image](https://github.com/GreaterLondonAuthority/dfl-ckanext/assets/1217533/47266b58-03db-455f-8991-9b468f591b25)

It'd probably be good to highlight relevance by default?

There's also a tiny bit of weirdness where the underline extends past the end of `Relevance` which would be good to get rid of.
![image](https://github.com/GreaterLondonAuthority/dfl-ckanext/assets/1217533/e225aa85-fdc1-40f6-b536-9a18bb9b0004)

![image](https://github.com/GreaterLondonAuthority/dfl-ckanext/assets/1217533/e5d0def3-2ff9-488c-b0ed-b3e462aed735)

madelinekosse committed:
Fix spacing of order-by links

madelinekosse committed:
Highlight default sort order

The default sorting order is "score desc, metadata_modified desc", which means
that it matches neither of the options so neither option is highlighted by
default. This change means the first sort order in the list is highlighted.

kjoshi reviewed (approved):
> The default sorting order is "score desc, metadata_modified desc", which means that it matches neither of the options

Presumably `score desc` is the same as Relevance, and I imagine `metadata_modified` then only comes into play if two scores are identical, which is pretty unlikely. 

So this seems Ok, thanks 👍
